sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2300,6 +2332,91 @@ def __getitem__(self, key):
     return self.attributes[key].__get__(self, WriteResult)
 
 
+def _default_io_expansion_service(append_args=None):
+  return BeamJarExpansionService(
+      'sdks:java:io:google-cloud-platform:expansion-service:build',
+      append_args=append_args)
+
+
+class StorageWriteToBigQuery(PTransform):
+  """Writes data to BigQuery using Storage API."""
+  URN = "beam:schematransform:org.apache.beam:bigquery_storage_write:v1"
+  FAILED_ROWS = "failed_rows"
+  FAILED_ROWS_WITH_ERRORS = "failed_rows_with_errors"
+
+  def __init__(
+      self,
+      table,
+      create_disposition="",
+      write_disposition="",
+      triggering_frequency=0,
+      use_at_least_once=False,
+      expansion_service=None):
+    """Initialize a StorageWriteToBigQuery transform.
+
+    :param table:
+      Fully-qualified table ID specified as ``'PROJECT:DATASET.TABLE'``.
+    :param create_disposition:
+      String specifying the strategy to take when the table doesn't
+      exist. Possible values are:
+      * ``'CREATE_IF_NEEDED'``: create if does not exist.
+      * ``'CREATE_NEVER'``: fail the write if does not exist.
+    :param write_disposition:
+      String specifying the strategy to take when the table already
+      contains data. Possible values are:
+      * ``'WRITE_TRUNCATE'``: delete existing rows.
+      * ``'WRITE_APPEND'``: add to existing rows.
+      * ``'WRITE_EMPTY'``: fail the write if table not empty.
+    :param triggering_frequency:
+      The time in seconds between write commits. Should only be specified
+      for streaming pipelines. Defaults to 5 seconds.
+    :param use_at_least_once:
+      Use at-least-once semantics. Is cheaper and provides lower latency,
+      but will potentially duplicate records.
+    :param expansion_service:
+      The address (host:port) of the expansion service. If no expansion
+      service is provided, will attempt to run the default GCP expansion
+      service.
+    """
+    super().__init__()
+    self._table = table
+    self._create_disposition = create_disposition
+    self._write_disposition = write_disposition
+    self._triggering_frequency = triggering_frequency
+    self._use_at_least_once = use_at_least_once
+    self._expansion_service = (
+        expansion_service or _default_io_expansion_service())
+    self.schematransform_config = SchemaAwareExternalTransform.discover_config(
+        self._expansion_service, self.URN)
+
+  def expand(self, input):
+    opts = input.pipeline.options.view_as(StandardOptions)
+    # TODO(https://github.com/apache/beam/issues/21307): Add support for
+    # OnWindowExpiration to more runners. Storage Write API requires
+    # `beam:requirement:pardo:on_window_expiration:v1` when unbounded
+    available_runners = ['DataflowRunner', 'TestDataflowRunner']
```

Comment:
I don't see the Java implementation limited to Dataflow this way. Why do we just want the limit the Python x-lang version ?